### PR TITLE
[prometheusremotewrite] fix: counter name check 

### LINKED ISSUE #2608: prometheusremotewrite exporter crashes collector with metrics names < 6 characters
**Describe the bug**
When Collector uses prometheusremotewriteexporter, metric names longer than 6 characters cause it to crash

2021-03-03 09:30:54
panic: runtime error: slice bounds out of range [-1:]

2021-03-03 09:30:54
goroutine 97 [running]:

2021-03-03 09:30:54
go.opentelemetry.io/collector/exporter/prometheusremotewriteexporter.getPromMetricName(0xc0003df7c0, 0x0, 0x0, 0x6, 0x6)

Seems to be caused by this logic which checks for "total" at the end of the metric name: https://github.com/open-telemetry/opentelemetry-collector/blob/68c5961f7bc2d8d55fd431ab07367fc16f8b26d0/exporter/prometheusremotewriteexporter/helper.go#L197

**Steps to reproduce**
Enable Prometheus Remote Write exporter, send OTLP metrics to collector with names shorter than 6 characters

**What version did you use?**
Version: 0.16.0

**What config did you use?**
Config:
extensions:
  health_check:
    port: 13133

receivers:
  otlp:
    protocols:
      grpc:
        endpoint: 0.0.0.0:4317
      http:

processors:
  batch:

exporters:
  logging:
    logLevel: warn
  prometheusremotewrite:
    endpoint: "http://127.0.0.1:9201/write"

service:
  pipelines:
    traces:
      receivers: [otlp]
      processors: [batch]
      exporters: [logging]

    metrics:
      receivers: [otlp]
      processors: [batch]
      exporters: [logging, prometheusremotewrite]

  extensions: [health_check]

**Environment**
OS: alpine:3.12 (Docker)

### PR DESCRIPTION
**Pull request by naseemkullah**
fixes  #2608

~Ensure that the metric name is greater than the length of the
counter suffix ("_total") before checking if it contains the counter
suffix to prevent crashes for short (smaller or eq to the length of the
suffix) metric names.~

~The edge case where the metric name is "_total" is not handled (considered
not to have the suffix), but should not occur IRL.~

edit: Use `strings.hasSuffix()` to check if metric name has the counter suffix ("_total")

Also fix the spelling of "delimiter".

warning: this is one of my first PRs in Go 😅

### CODE REVIEW

File: exporter/prometheusremotewriteexporter/helper.go [hunk 1]
```diff
@@ -205,9 +202,12 @@ func getPromMetricName(metric *otlp.Metric, ns string) string {
 		}
 	*/
 
-	if b.Len() > 0 && isCounter {
-		b.WriteString(delimeter)
-		b.WriteString(totalStr)
+	counterSuffixLength := len(counterSuffix)
+	nameLength := len(name)
+	hasCounterSuffix := nameLength > counterSuffixLength && name[nameLength-counterSuffixLength:] == counterSuffix
```
Comment (bogdandrutu) [inline exporter/prometheusremotewriteexporter/helper.go:207]: `strings.HasSuffix(name, counterSuffix)`?

File: exporter/prometheusremotewriteexporter/helper.go [hunk 2]
```diff
@@ -205,9 +202,12 @@ func getPromMetricName(metric *otlp.Metric, ns string) string {
...
+
+	if b.Len() > 0 && isCounter && !hasCounterSuffix {
```
Comment (bogdandrutu) [inline exporter/prometheusremotewriteexporter/helper.go:209]: `if b.Len() > 0 && isCounter && !strings.HasSuffix(name, counterSuffix) {`

Comment (naseemkullah) [reply exporter/prometheusremotewriteexporter/helper.go]: nice, thanks

**Review by bogdandrutu — APPROVED** [review]

**Review by bogdandrutu — APPROVED** [review]
Would be good to have a test :)

**Comment by naseemkullah** [comment]
> Would be good to have a test :)

done, PTAL -> https://github.com/open-telemetry/opentelemetry-collector/pull/2613/commits/3bd2bd7258fb086b3f44fe410b1467684db4e0fd

**Pull request merged**